src/openassetio-python/cmodule/src/BatchElementErrorBinding.cpp

```diff
@@ -24,6 +25,8 @@ void registerBatchElementError(py::module &mod) {
   batchElementError
       .def(py::init<BatchElementError::ErrorCode, openassetio::Str>(), py::arg("code"),
            py::arg("message"))
+      .def(py::self == py::self)
```

Comment:
[fix] Would be good to have a little `pytest` unit test for `==` and `!=` .

Reply:
As we have an explicit implementation of `!=` which we know isn't covered by `pytest` would be good to have a cpp test case too?

Reply:
C++ done, thanks: https://github.com/OpenAssetIO/OpenAssetIO/pull/881/commits/d3692d300f310c58624c523552c86c8b3d6a09f3

Reply:
We still don't have explicit `pytest` tests for the equality operator. However I'm not so worried, since it's tested implicitly by the fact we use it in other tests.

Reply:
As we have to explicitly implement it on the python side, can we make sure we have python coverage too? Its quite a core object and we have coverage for other classes.

Reply:
Added in https://github.com/OpenAssetIO/OpenAssetIO/pull/881/commits/718a813119f6d70d3fb06e6ae87206f71d95de52 - thanks!

Reply:
Awesome, thanks!